test/Tests/src/Data/Text_Spec.enso

```diff
@@ -274,7 +274,55 @@ spec =
             utf_8_whitespace.split "\s+" use_regex=True . should_equal utf_8_whitespace_split
             'abc  def\tghi'.split '\\s+' use_regex=True . should_equal ["abc", "def", "ghi"]
 
-        Test.specify "exmples should be correct" <|
+        Test.specify 'should be able to split with a vector of strings' <|
+            'azbzczdzezfzg'.split ['b', 'zez'] . should_equal ['az', 'zczd', 'fzg']
+            'blah x 123'.split [' ', ' x ' , 'x'] . should_equal ['blah', '', '', '123']
```

Comment:
Hmm, I just noticed this case and it is interesting. Maybe we should specify what happens when multiple delimiters overlap?


My intuition is it may be best to yield `['blah', '123']` here - i.e. take the 'largest' match. But I'm not sure how much that complicates stuff - so not suggesting to necessarily implement it - just wondering what is a good approach.

I'd consider adding one more test like this:
```
'abcdef'.split ['bc', 'cd'] == ???
```

I wonder what the answer will be? I guess probably `['a', 'def']`? Or is it `['a', 'ef']`? Both seem to make some sense (first that it matches the first delimiter, second that it does a 'union of all matched delimiter sections').

I'd appreciate adding this test, just to sanity check that we are not getting anything unexpected here (for example, I'd not expect to have any `''` in the result here).

Reply:
The handling as Greg has is what I specified with him. 

![image](https://user-images.githubusercontent.com/4699705/229780583-2f46cc67-9149-48df-9663-72b81549d4f9.png)

It is what happens if you put the sequence into a regex split (i.e. split on `( | x |x)`. It is a bit of a degenerate case but is reasonable response.

Reply:
Fair. Still, I'd appreciate the `'abcdef'.split ['bc', 'cd']` test just to 'document' the behaviour, if we may.

Reply:
Added more test cases.